src/adapters/source-adapters/BoundedUnionSourceAdapter.sol

```diff
@@ -91,21 +104,46 @@ abstract contract BoundedUnionSourceAdapter is
         override(ChainlinkSourceAdapter, ChronicleMedianSourceAdapter, PythSourceAdapter)
         returns (int256, uint256, uint256)
     {
-        // Chainlink has price history, so use tryLatestDataAt to pull the most recent price that satisfies the timestamp constraint.
-        (int256 clAnswer, uint256 clTimestamp,) = ChainlinkSourceAdapter.tryLatestDataAt(timestamp, maxTraversal);
+        // In the happy path there have been no source updates since requested time, so we can return the latest data.
+        AllSourceData memory data = _getAllLatestSourceData();
+        (int256 boundedAnswer, uint256 boundedTimestamp) = _selectBoundedPrice(
```

Comment:
Can we do `_selectBoundedPrice(data)`

and update _selectBoundedPrice like
```
    function _selectBoundedPrice(AllSourceData memory data)
        internal
        view
        returns (int256, uint256)
    {
```

Wdyt?

Reply:
great suggestion, pushing commit for this